File: scatterplotMatrix.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from pandas.api.types import is_numeric_dtype
from scipy.stats import pearsonr


# Bring dataset into memory
iris = pd.read_csv(r"G:\My Drive\iris.csv")

# Verify the structure of the data
iris.head(10)
iris.info()

# Make any necessary datatype changes
iris["Sepal.Length"] = iris["Sepal.Length"].astype(float)

# Isolate only those columns which are numeric/continuous
numerics = ['int16', 'int32', 'int64', 'float16', 'float32', 'float64']
#iris = iris.select_dtypes(include=numerics)

# Output Scatterplot Version 1
def reg_coef(x,y,label=None,color=None,**kwargs):
    ax = plt.gca()
    r,p = pearsonr(x,y)
    ax.annotate('r = {:.2f}'.format(r), xy=(0.5,0.5), \
                xycoords='axes fraction', ha='center')
    ax.set_axis_off()

g = sns.PairGrid(iris, hue="Species")
g.map_diag(sns.distplot)
g.map_lower(sns.regplot)
g.map_upper(reg_coef)


# Output Scatterplot Version 2
g = sns.PairGrid(iris, hue="Species")
g = g.map_diag(plt.hist)
g = g.map_offdiag(plt.scatter)
g = g.add_legend()

File: scatterplotMatrix.R
library(ggplot2)
library(GGally)

# Bring dataset into memory
attach(iris)

# Verify the structure of the data
head(iris, 10)
str(iris)

# Make any necessary datatype changes
iris$Sepal.Length <- as.numeric(iris$Sepal.Length)

# Isolate only those columns which are numeric/continuous
num_only <- unlist(lapply(iris, is.numeric))

# Create regression line function
panel.lm <- function(x, y, pch=par("pch"), col.lm="red", ...){
  ymin <- min(y)
  ymax <- max(y)
  xmin <- min(x)
  xmax <- max(x)
  ylim <- c(min(ymin,xmin),max(ymax,xmax))
  xlim <- ylim
  points(x, y, pch = pch,ylim = ylim, xlim= xlim,...)
  ok <- is.finite(x) & is.finite(y)
  if (any(ok)) 
    abline(lm(y[ok]~ x[ok]), 
           col = col.lm, ...)
}

# Output Scatterplot Version 1
pairs(iris[,num_only], panel = panel.lm)

# Output Scatterplot Version 2
ggpairs(iris[,num_only], ggplot2::aes(colour=iris$Species),  panel = panel.lm)
      